keps/sig-node/3721-support-for-env-files/README.md

```diff
@@ -0,0 +1,570 @@
+we are assuming the cluster may have kubelets with older versions (without
+this KEP' changes):
+
+1. When the kubelet is upgraded, the env files will be instantiated in the
+   container. On downgrade, the env files will be ignored but the pod will still
+   run. On upgrade, the env files should be instantiated in the container again.
+
+2. When the apiserver is upgraded, new envs will be written. But on downgrade,
+   it cannot be written but the existing envs will continue to exist. On
+   upgrade, the envs can again be written.
+
+###### Is the rollout accompanied by any deprecations and/or removals of features, APIs, fields of API types, flags, etc.?
+
+No
+
+### Monitoring Requirements
+
+###### How can an operator determine if the feature is in use by workloads?
+
+The feature is built into the kubelet and api server components. No metric is
+planned at this moment. The way to determine usage is by checking whether the
+pods/containers have a `FileEnvSource` or `FileKeySelector` fields set.
+
+###### How can someone using this feature know that it is working for their instance?
+
+The user can exec into the running container and check that the environment
+variable is set:
+
+```
+$ kubectl exec -n $NAMESPACE $POD_NAME -- env
+CONFIG_VAR=HELLO
+```
+
+###### What are the reasonable SLOs (Service Level Objectives) for the enhancement?
+
+The startup latency of schedulable stateless/stateful Pods may increase, but will remain consistent with the existing SLO standard.
+
+ref: https://github.com/kubernetes/community/blob/master/sig-scalability/slos/slos.md#steady-state-slisslos
+
+
+###### What are the SLIs (Service Level Indicators) an operator can use to determine the health of the service?
+
+We can determine whether the service is healthy by checking if the `started_containers_errors_total` and `started_pods_errors_total` metrics are abnormally increasing.
+
+###### Are there any missing metrics that would be useful to have to improve observability of this feature?
+
+No
+
+### Dependencies
+
+###### Does this feature depend on any specific services running in the cluster?
+
+No
+
+### Scalability
+
+###### Will enabling / using this feature result in any new API calls?
+
+No
+
+###### Will enabling / using this feature result in introducing new API types?
+
+No
+
+###### Will enabling / using this feature result in any new calls to the cloud provider?
+
+No
+
+###### Will enabling / using this feature result in increasing size or count of the existing API objects?
+
+We have added the `fileKeyRef` data structure to `podSpec`, which will undoubtedly increase the size of the Pod API. The increase in size depends on the number of environment variables defined by the user, making it impossible to estimate the exact number of bytes added. However, this should not have a significant impact on the API. For many users, it merely involves migrating environment variables previously defined in `ConfigMap/Secret` to this new structure.
```

Comment:
I'll let this go for alpha, but for beta this will need a proper estimation, like all the other features.